xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/document/UserCache.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.user.internal.document;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.apache.commons.lang3.function.FailableFunction;
+import org.xwiki.cache.Cache;
+import org.xwiki.cache.CacheException;
+import org.xwiki.cache.CacheManager;
+import org.xwiki.cache.config.LRUCacheConfiguration;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLifecycleException;
+import org.xwiki.component.phase.Disposable;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.model.reference.WikiReference;
+
+/**
+ * Cache various user related information.
+ * 
+ * @version $Id$
+ * @since 17.3.0RC1
+ */
+@Component(roles = UserCache.class)
+@Singleton
+public class UserCache implements Initializable, Disposable
+{
+    @Inject
+    private CacheManager cacheManager;
+
+    private Cache<Boolean> hasUserCache;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        try {
+            this.hasUserCache = this.cacheManager.createNewCache(new LRUCacheConfiguration("use.hasUsers"));
```

Comment:
typo in the cache name: I guess it should be `user.hasUsers`

Reply:
Indeed.